### ISSUE TITLE
Event Log / Alerts

### ISSUE THREAD

**Issue by rainer-tammer** [sni/Thruk#732]
Hello,
The event log and the alert log are still not working.

Software:

thruk 2.14-2
icinga2 2.6.3-1

OS: SLES 11 SP4

The connection is via licestatus (/var/run/icinga2/cmd/livestatus).

Is there s solution to see content in the event/alert log?
A monitoring tool without the logs is a big problem.....

Bye
  Rainer Tammer

**Comment by sni** [comment]
See https://thruk.org/documentation/logfile-cache.html#icinga-2
You can enable the logcache as a workaround. Besides that, this need to be fixed in Icinga 2.

**Comment by rainer-tammer** [comment]
Hello,
One additional note:

I can see events if I enter "*" as search pattern in the Event Log view.
There is also an error message displayed:

invalid regular expression: Quantifier follows nothing in regex
marked by <-- HERE in
* <-- HERE 

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
Please hold this ticket for 2 days.
After waiting for 1h some log messages showed up... 
This sometimes happened with the older releases but did not work for a longer time.

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
I had a discussion about this topic with the icinga2 developers last year -> no problem/won't fix... :-(.

I have installed icinga2 on SLES 11 with PostgreSQL, so I will have to install an additional MySQL DB, right? Is there a script to create the necessary tables?

Bye
  Rainer

**Comment by sni** [comment]
The tables will be created automatically.

**Comment by rainer-tammer** [comment]
Hello,
Can I enable this manually without omd?

I have installed MySQL but I get an error:

icinga01:/etc/thruk # thruk -a logcacheimport --local
import removes current cache and imports new logfile data.
use logcacheupdate to update cache. Continue? [n]: y
running import for site icinga01
[12:50:16,926][INFO][Thruk] logcache import failed: something went wrong, cannot get start/end from logfiles (0 / 1499337955) at /usr/share/thruk/lib/Thruk/Backend/Provider/Mysql.pm line 1582.
ERROR: something went wrong, cannot get start/end from logfiles (0 / 1499337955) at /usr/share/thruk/lib/Thruk/Backend/Provider/Mysql.pm line 1582.

ERROR - imported 0 log items from 1 site with 1 errors in 3.99s (0/s)

-----------------------

icinga01:/etc/thruk # thruk -a logcachestats
FAILED - logcache is not enabled

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
Note: I have the problem that icinga2 does not correctly rotate the logs in /var/log/icinga2/compat...

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
Please ignore the message (I did not restart thruk prior to running the command):

icinga01:/etc/thruk # thruk -a logcachestats
FAILED - logcache is not enabled

I still have the problem that the import fails.
The 1h logrotate in icinga2 is not working, this will probably render this method useless, right?
The icinga2 developers wote last year that the /var/log/icinga2/compat is old/depreciated.

Bye
  Rainer

**Comment by sni** [comment]
"logcacheimport" uses livestatus, which is broken in Icinga 2. Please read https://thruk.org/documentation/logfile-cache.html#icinga-2 It's not that much text. You need to import the logfiles directly.

**Comment by rainer-tammer** [comment]
Hello,
OK, I could import the log entries now, but I still have two (major) problems:

1. The compat log is depreciated according to the icinga2 developers
2. The log rotation does not work (do you know this problem?)

See:

icinga01:/var/log/icinga2/compat # ls -al
total 48
drwxr-x--- 3 icinga icingacmd    38 Jul  6 11:59 .
drwxr-x--- 4 icinga icingacmd  4096 Jul  6 12:57 ..
drwxr-x--- 2 icinga icingacmd  4096 Jul  6 11:59 archives
-rw-r--r-- 1 icinga icinga    38778 Jul  6 15:53 icinga.log


icinga01:/var/log/icinga2/compat # ls -al archives
total 15728
drwxr-x--- 2 icinga icingacmd    4096 Jul  6 11:59 .
drwxr-x--- 3 icinga icingacmd      38 Jul  6 11:59 ..
-rw-r--r-- 1 icinga icinga    1463008 Feb 17 16:59 icinga-02-17-2017-16.log
-rw-r--r-- 1 icinga icinga    1994334 Apr 25 16:28 icinga-04-25-2017-16.log
-rw-r--r-- 1 icinga icinga     187110 Apr 28 08:48 icinga-04-28-2017-08.log
-rw-r--r-- 1 icinga icinga       6756 May 30  2016 icinga-05-30-2016-13.log
-rw-r--r-- 1 icinga icinga    1635729 May 30 11:58 icinga-05-30-2017-11.log
-rw-r--r-- 1 icinga icinga      13904 May 31  2016 icinga-05-31-2016-09.log
-rw-r--r-- 1 icinga icinga      46219 May 31  2016 icinga-05-31-2016-13.log
-rw-r--r-- 1 icinga icinga     284278 May 31 15:36 icinga-05-31-2017-15.log
-rw-r--r-- 1 icinga icinga      69789 Jun  2  2016 icinga-06-02-2016-10.log
-rw-r--r-- 1 icinga icinga     183124 Jun  2  2016 icinga-06-02-2016-11.log
-rw-r--r-- 1 icinga icinga      61025 Jun  2  2016 icinga-06-02-2016-12.log
-rw-r--r-- 1 icinga icinga      44710 Jun  2  2016 icinga-06-02-2016-13.log
-rw-r--r-- 1 icinga icinga      94943 Jun  3  2016 icinga-06-03-2016-09.log
-rw-r--r-- 1 icinga icinga     163175 Jun  3  2016 icinga-06-03-2016-11.log
-rw-r--r-- 1 icinga icinga      52832 Jun  3  2016 icinga-06-03-2016-12.log
-rw-r--r-- 1 icinga icinga      38961 Jun  3  2016 icinga-06-03-2016-15.log
-rw-r--r-- 1 icinga icinga     295687 Jun  6  2016 icinga-06-06-2016-10.log
-rw-r--r-- 1 icinga icinga      83015 Jun  7  2016 icinga-06-07-2016-10.log
-rw-r--r-- 1 icinga icinga     116765 Jun 10  2016 icinga-06-10-2016-08.log
-rw-r--r-- 1 icinga icinga     134736 Jun 15  2016 icinga-06-15-2016-10.log
-rw-r--r-- 1 icinga icinga     230463 Jun 24  2016 icinga-06-24-2016-23.log
-rw-r--r-- 1 icinga icinga    2389686 Jul  6 11:57 icinga-07-06-2017-11.log
-rw-r--r-- 1 icinga icinga     899999 Jul 11  2016 icinga-07-11-2016-23.log
-rw-r--r-- 1 icinga icinga     414851 Jul 21  2016 icinga-07-21-2016-23.log
-rw-r--r-- 1 icinga icinga    4338727 Sep 14  2016 icinga-09-14-2016-23.log
-rw-r--r-- 1 icinga icinga      97738 Nov 23  2016 icinga-11-23-2016-16.log
-rw-r--r-- 1 icinga icinga     670965 Dec 23  2016 icinga-12-23-2016-12.log

The log should be rotated every hour. This does work after stopping/starting icinga2.

[1499335200] LOG ROTATION: HOURLY
[1499335200] LOG VERSION: 2.0
[1499335200] CURRENT HOST STATE: xxxx;UP;HARD;1;PING OK - Packet loss = 0%, RTA = 210.26 ms
....

------[compatlog.conf]----------------------------
/**
 * The CompatLogger type is responsible for writing log files in a format
 * that is compatible with Icinga 1.x.
 */

library "compat"

object CompatLogger "compatlog" {
  rotation_method = "HOURLY"
}
------[compatlog.conf]----------------------------

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
Please have a look at:

https://github.com/Icinga/icinga2/issues/4267
>
>Updated by mfriedrich on 2016-12-07 23:26:03 +00:00
>
>    Status changed from New to Closed
>
>I've had a discussion with Michael Kraus during the OSMC hackathon on how Thruk stores the historical >data for reporting and such. There are other issues with timestamps inside an HA cluster which would >require creating a log indexer similar to Elasticsearch (which we are not going to implement either). Their >current approach is to use the IDO database table "log_entries" with an external daemon then dumping >that to the Thruk MySQL database for reporting.
>
>My suggestion would be to even enhance that and directly rely on the statehistory and >contactnotifications tables as Icinga Web 2 does. Or using the API event stream dumping those events >directly into the Thruk database. Fetching the compat log via Livestatus is really broken by design, and >something we will not fix, unfortunately. Unless someone comes up with a patch on their own.
>
>Kind regards,
>Michael

So we do currently have no really working solution, right?

Bye
  Rainer

**Comment by rainer-tammer** [comment]
Hello,
One idea: All icinga2 log entries are in the (PostgreSQL) table icinga_logentries.
I do not know the code of thruk very well, but would it be possible to directly read the log entries fromj this table?

Bye
  Rainer

**Comment by sni** [comment]
I assume that table is as broken, unreliable and deprecated as the compat log itself.
The logrotation issue must be fixed in Icinga2.

**Comment by rainer-tammer** [comment]
Hello,
OK, than we have a problem. I can currently see no reliable way to get the events/alerts in thruk. I really like the thruk GUI, but without reliable events/alerts.... :-(.

icinga2: wont fix (https://github.com/Icinga/icinga2/issues/4267)
thruk: can't fix because it's an icinga2 problem.

Any recommendations?
Would it be better to switch back to icinga1?

Do you know a icionga2 developer which I could ask?

Bye
  Rainer

**Comment by sni** [comment]
Well, i'd like to have a reliable source of log events in icinga2 as well, but this is the Thruk issue tracker and there is nothing i can do about it. And yes, this is one of the reasons we don't use Icinga 2 in production yet.
The workaround with the logcache and imports from file should work so far till Icinga 2 comes up with an reliable log source.

**Comment by rainer-tammer** [comment]
Hello,
OK, I can try to open a new ticket @icinga2.

What would be the preferred icinga2 log implementation?

Bye
  Rainer

**Comment by sni** [comment]
Anythings works, extra points for a single access point, even when using clustered mode.
